Assets/Plugins/LeapMotion/Core/Editor/LeapServiceProviderEditor.cs

```diff
@@ -61,79 +80,205 @@ public class LeapServiceProviderEditor : CustomEditorBase<LeapServiceProvider> {
+        // it and will then flip them again. There appears to be no way to clear this cache, so here we
+        // check the first normal in the mesh to see if we need to flip it. 
+       // if (_rigelInteractionZoneMesh.normals[0].ApproxEquals(new Vector3(-1.0f, 0.1f, 0)))
+          ReverseNormals();
+      }
+
+      if (_rigelInteractionMaterial == null) {
+        _rigelInteractionMaterial = (Material)AssetDatabase.LoadAssetAtPath(Path.Combine("Assets", "UnityModules", "Assets", "Plugins", "LeapMotion", "Core", "Materials", "RigelInteractionVolume.mat"), typeof(Material));
+      }
+
+    }
+
+    private void ReverseNormals() {
+      Vector3[] normals = _rigelInteractionZoneMesh.normals;
+      for (int i = 0; i < normals.Length; i++)
+        normals[i] = -normals[i];
+      _rigelInteractionZoneMesh.normals = normals;
+    }
+
+    private LeapServiceProvider LeapServiceProvider {
+      get {
+
+        if (this._leapServiceProvider != null) { 
+          return this._leapServiceProvider;
+        }
+        else {
+          this._leapServiceProvider = this.target.GetComponent<LeapServiceProvider>();
+
+          return this._leapServiceProvider;
+        }
+      }
+    }
+
+    private Controller LeapController {
+      get {
+
+        if (this._leapController != null) {
+          return this._leapController;
+        }
+        else
+        {
+          this._leapController = LeapServiceProvider?.GetLeapController();
+
+          if (this._leapController != null) {
+            this._leapController.Device += _leapController_DeviceChanged;
+            this._leapController.DeviceLost += _leapController_DeviceChanged;
+          }
+
+          return this._leapController;
+        }
+      }
+    }
+
+    private void _leapController_DeviceChanged(object sender, DeviceEventArgs e) {
+      EditorWindow view = EditorWindow.GetWindow<SceneView>();
+      view.Repaint();
+    }
+
+    private void DetectConnectedDevice() {
+
+      if (LeapController?.Devices?.Count == 1)
+      {
+        if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_RIGEL) {
+          DrawRigelInteractionZoneMesh();
+        }
+        else if (LeapController.Devices.First().Type == Device.DeviceType.TYPE_PERIPHERAL) {
+          DrawLeapMotionControllerInteractionZone(LMC_BOX_WIDTH, LMC_BOX_DEPTH, LMC_BOX_RADIUS, Color.white);
+        }
+      }
+    }
+
+    private LeapServiceProvider.InteractionVolumeVisualization? GetSelectedInteractionVolume() {
+
+      return LeapServiceProvider?.SelectedInteractionVolumeVisualization;
+    }
+
+    private void DrawRigelInteractionZoneMesh() {
+
+      if (_rigelInteractionMaterial != null && _rigelInteractionZoneMesh != null) {
+        _rigelInteractionMaterial.SetPass(0);
+        Graphics.DrawMeshNow(_rigelInteractionZoneMesh, Matrix4x4.TRS(_rigelInteractionZoneMeshOffset, Quaternion.Euler(-90, 0, 0), Vector3.one * 0.001f));
```

Comment:
The transform which the mesh is drawn at is always at the origin.  So if the ServiceProvider moves, the mesh will stay behind.

You can fix this by replacing `Matrix4x4.TRS(_rigelInteractionZoneMeshOffset, Quaternion.Euler(-90, 0, 0), Vector3.one * 0.001f)` with `target.transform.localToWorldMatrix * Matrix4x4.TRS(_rigelInteractionZoneMeshOffset, Quaternion.identity, Vector3.one * 0.001f)`.  This will move the interaction volume to always be in front of the service provider.